addons/stock_account/models/stock_quant.py

```diff
@@ -0,0 +1,45 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+
+
+class StockQuant(models.Model):
+    _inherit = 'stock.quant'
+
+    value = fields.Float('Value', compute='_compute_value')
+    currency_id = fields.Many2one(related='product_id.currency_id')
+
+    @api.depends('quantity')
+    def _compute_value(self):
+        """ For standard and AVCO valuation, compute the current accounting
+        valuation of the quants by multiplying the quantity by
+        the standard price. Instead for FIFO, use the quantity times the
+        average cost (valuation layers are not manage by location so the
+        average cost is the same for all location and the valuation field is
+        a estimation more than a real value).
+        """
+        for quant in self:
+            if quant.product_id.cost_method == 'fifo':
+                quantity = quant.product_id.quantity_svl
+                if not quantity:
+                    quant.value = 0.0
+                average_cost = quant.product_id.value_svl / quantity
```

Comment:
potentiel div by zero in `average_cost = quant.product_id.value_svl / quantity
`
```python
if not quantity:
     quant.value = 0.0
     continue
```